metal/eip_controlplane_reconciliation.go

```diff
@@ -267,106 +287,273 @@ func (m *controlPlaneEndpointManager) reassign(ctx context.Context, nodes []*v1.
 	return errors.New("ccm didn't find a good candidate for IP allocation. Cluster is unhealthy")
 }
 
-// reconcileServices ensure that our Elastic IP is assigned as `externalIPs` for
-// the `default/kubernetes` service
-func (m *controlPlaneEndpointManager) reconcileServices(ctx context.Context, svcs []*v1.Service, mode UpdateMode) error {
+func isControlPlaneNode(node *v1.Node) bool {
+	for _, label := range controlPlaneLabels {
+		if metav1.HasLabel(node.ObjectMeta, label) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (m *controlPlaneEndpointManager) nodeFilter(n *v1.Node) bool {
```

Comment:
Can we add a comment here what this filter does? Specifically that it:

* applies only to control plane nodes
* only if the `eipTag` is set
* only if the `apiServerPort` has been discovered (i.e. might be too early)

Reply:
I removed the function definition, moving the logic to where we are defining the EventHandler. I also added additional comments there to help clarify